cpp/src/parquet/encryption/internal_file_decryptor.cc

```diff
@@ -64,17 +66,20 @@ InternalFileDecryptor::InternalFileDecryptor(FileDecryptionProperties* propertie
   properties_->set_utilized();
 }
 
+InternalFileDecryptor::~InternalFileDecryptor() { WipeOutDecryptionKeys(); }
+
 void InternalFileDecryptor::WipeOutDecryptionKeys() {
-  std::lock_guard<std::mutex> lock(mutex_);
+  std::unique_lock lock(mutex_);
```

Comment:
This method is only called from the destructor, so there is no need for a lock, as its guaranteed to be called only once.

Reply:
I guess the reason is that `WipeOutDecryptionKeys()` might be called in the `ParquetFileReader::Close()`